tests/test_utils.py

```diff
@@ -313,3 +313,16 @@ def test_check_status_code_for_missing_status_code(capsys, repo_url):
 
     captured = capsys.readouterr()
     assert captured.out == "NOTE: Try --verbose to see response content.\n"
+
+
+@pytest.mark.parametrize(
+    ("num_bytes, expected"),
+    [(3704, "3.6 KB"), (1153433, "1.1 MB"), (21412841, "20.4 MB")],
+)
+def test_get_file_size(num_bytes, expected, monkeypatch):
+    """Get the size of file as a string with units."""
+    monkeypatch.setattr(os.path, "getsize", lambda num_bytes: num_bytes)
```

Comment:
The argument to `getsize` is a filename, but a common pattern (that I think avoids linting errors) is to use an underscore for unused arguments:

```suggestion
    monkeypatch.setattr(os.path, "getsize", lambda _: num_bytes)
```